azurerm/internal/services/mssqlvm/resource_arm_mssql_virtual_machine.go

```diff
@@ -0,0 +1,495 @@
+package mssqlvm
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"log"
+
+	"github.com/Azure/azure-sdk-for-go/services/preview/sqlvirtualmachine/mgmt/2017-03-01-preview/sqlvirtualmachine"
+	"github.com/hashicorp/go-azure-helpers/response"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/features"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tags"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmMsSqlVirtualMachine() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmMsSqlVirtualMachineCreateUpdate,
+		Read:   resourceArmMsSqlVirtualMachineRead,
+		Update: resourceArmMsSqlVirtualMachineCreateUpdate,
+		Delete: resourceArmMsSqlVirtualMachineDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"resource_group_name": azure.SchemaResourceGroupNameDiffSuppress(),
+
+			"location": azure.SchemaLocation(),
+
+			"virtual_machine_resource_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: azure.ValidateResourceID,
+			},
+
+			"sql_license_type": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(sqlvirtualmachine.PAYG),
+					string(sqlvirtualmachine.AHUB),
+				}, false),
+			},
+
+			"sql_sku": {
+				Type:     schema.TypeString,
+				Optional: true,
```

Comment:
I communicated with sql vm team, they think this field is not necessary. Because the extension will figure out what version is installed if you don't provide it. So I remove it.